frappe/public/js/frappe/list/list_view_select.js

```diff
@@ -191,12 +191,16 @@ frappe.views.ListViewSelect = class ListViewSelect {
 			);
 		});
 
-		this.page.add_custom_menu_item(
-			kanban_switcher,
-			__("Create New Kanban Board"),
-			() => frappe.views.KanbanView.show_kanban_dialog(this.doctype),
-			true
-		);
+		let perms = this.list_view.board_perms;
```

Comment:
Need Kanban Board Doctype permissions here, not Kanban Board document permissions